### PR TITLE
Split interaction with API from client to create an SDK.

### LINKED ISSUE #2280: RPC Abstraction should be usable by external projects
**Is your feature request related to a problem? Please describe.**
The client use a great interface to communicate with the RPC API. If I want to develop a tool for massa and I want to use the RPC API I would like to use this interface but it's in a binary not a library. So I have to rewrite mine (or copy paste).

**Describe the solution you'd like**
The solution would be to split the interactions with the RPC API to make it a separate crate as a library and publish it so that everyone can use it as a base to code their own tool in Rust.

**Describe alternatives you've considered**
/

**Additional context**
/

### PR DESCRIPTION
**Pull request by AurelienFT**
Fix #2280 

### Motivations

Currently, if someone wants to develop a tool on massa he will have to create his own interactions with the API but we already have one that we use for the client. I think we could make those interactions as a new package that can be consider as a SDK.

### Follow-up questions

I think it's a package to be published so that people can directly refer to it easily. Two problems I see with the publish : 
- Need a better documentation
- What about the versioning ? Should we have a version for each testnet ? Auto update on CI ?

### CODE REVIEW

**Review by yvan-sraka — DISMISSED** [review]
I'm not sure of the name `massa-sdk`, otherwise LGTM!

File: massa-sdk/src/lib.rs [hunk 1]
```diff
@@ -172,36 +176,30 @@ impl RpcClient {
...
     /// Get the block graph within the specified time interval.
     /// Optional parameters: from <time_start> (included) and to <time_end> (excluded) millisecond timestamp
-    pub(crate) async fn _get_graph_interval(
+    pub async fn _get_graph_interval(
         &self,
         time_interval: TimeInterval,
     ) -> RpcResult<Vec<BlockSummary>> {
         self.call_method("get_graph_interval", "Vec<BlockSummary>", time_interval)
             .await
     }
```
Comment (adrien-zinger) [inline massa-sdk/src/lib.rs:192]: Not sure. Maybe functions like that doesn't have to begin with a _ if it's pub ?

**Review by adrien-zinger — COMMENTED** [review]
We can talk about versioning yes and also about a design more "auto-generated" with proc macros.

I didn't test but it looks great!

File: massa-sdk/src/lib.rs (hunk 1 above)
Comment (AurelienFT) [reply massa-sdk/src/lib.rs]: Agree. Just made the modification.

**Comment by yvan-sraka** [comment]
I think the two interesting question @AurelienFT tackled here are: 

- how do we version API?

A quick answer would be using crate versioning of both API and SDK that would not be the same number as TestNet version (since a Protocol upgrade for example de not necessarily break the API contract).

- do we start uploading pieces of our software on crates.io to let community (and core team) more easily hack with it?

I have in mind of course the client and wallet crates!

What do you think about it @damip?

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
> * Need a better documentation

The api part is quite documented in the wiki, but the client is only documented in the help. Do you have any idea where it would be better to document it ?

Most command are only using an api endpoint

**Comment by AurelienFT** [comment]
bors merge

**Pull request merged**